Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -471,8 +471,13 @@ public final class PackageInfoMapper: PackageInfoMapping {
             let dependencyNames = target.dependencies.map(\.name)
             for dependencyName in dependencyNames {
                 let dependencyProducts = targetToProducts[dependencyName] ?? Set()
-                let destinations = unionDestinationsOfProducts(dependencyProducts, in: productDestinations)
-                testDestinations.formIntersection(destinations)
+                let isExternalDependency = dependencyProducts.isEmpty
```

Comment:
I don't 100 % follow why we can use `dependencyProducts.isEmpty` to determine whether a target is an external dependency. It would be amazing if we could have something more explicit.

Reply:
I think assigning an empty `Set` was making it unclearly. I've removed the ambiguous handling for external dependencies to make it explicit. Plus, it's simpler! How does it feel?